File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int i =0;
        int j = 0;
        int k = 0;
        int m = nums1.size();
        int n = nums2.size();

        int ans[m+n];

        while(i<m&&j<n){
            if(nums1[i]< nums2[j]){
                ans[k++] = nums1[i++];
            }else{
                ans[k] = nums2[j];
                k++;
                j++;
            }
        }

        while(i<m){
            ans[k++] = nums1[i++];
        }

        while(j<n){
            ans[k++] = nums2[j++];
        }

        int len = m+n;
        int mid = (len/2);
        if(len%2==0){
            return ((double)(ans[mid] + ans[mid-1])/2);
        }

        return ans[mid];
    }
};

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function(nums1, nums2) {
    let i = 0;
    let j = 0;
    let k = 0;
    let ans = []
    let m = nums1.length;
    let n = nums2.length;

    while(i<m && j<n){
        if(nums1[i] < nums2[j]){
            ans[k] = nums1[i];
            i++;
            k++;
        }else{
            ans[k] = nums2[j];
            k++;
            j++;
        }
    }
    while(i<nums1.length){
        ans[k] = nums1[i];
        i++;
        k++;
    }
    while(j<nums2.length){
        ans[k] = nums2[j];
        k++
        j++;
    }
    
    let len = ans.length;
    let mid = Math.floor(len/2);
    let midele = ans[mid]
    if(len%2==0){
      return  (ans[mid] + ans[mid-1])/2;
    }

   return ans[mid]
};